### PR TITLE
fix the package_url issue

### PR DESCRIPTION
**Pull request by Junefen**
- Define a new function in base named `virtwho_package_url`
- Update case package/test_install_uninstall.py/test_install_uninstall_by_rpm to use the the new function
- Update case package/test_upgrade_downgrade.py/test_install_uninstall_by_rpm to use the the new function 

```
% pytest --disable-warnings -v tests/package/ -k 'by_rpm'
========================================================= test session starts =========================================================
platform darwin -- Python 3.11.2, pytest-7.2.2, pluggy-1.0.0 -- /opt/homebrew/opt/python@3.11/bin/python3.11
cachedir: .pytest_cache
rootdir: /Users/yuefliu/workspace/virtwho-test, configfile: pytest.ini
collected 11 items / 9 deselected / 2 selected                                                                                        

tests/package/test_install_uninstall.py::TestInstallUninstall::test_install_uninstall_by_rpm PASSED                             [ 50%]
tests/package/test_upgrade_downgrade.py::TestUpgradeDowngrade::test_upgrade_downgrade_by_rpm PASSED                             [100%]

======================================= 2 passed, 9 deselected, 2 warnings in 97.42s (0:01:37) ========================================
```

### CODE REVIEW

File: tests/function/test_service.py [hunk 1]
```diff
@@ -134,9 +134,7 @@ def test_virtwho_service_control_by_ssh_connect(self, virtwho, ssh_guest, ssh_ho
         assert status == "dead"
 
         # start
-        ssh_guest.runcmd(
-            f"ssh {server} -p {port} " f'"systemctl restart virt-who"'
-        )
+        ssh_guest.runcmd(f"ssh {server} -p {port} " f'"systemctl restart virt-who"')
```
Comment (hkx303) [inline tests/function/test_service.py:137]:
```suggestion
        ssh_guest.runcmd(f"ssh {server} -p {port} systemctl restart virt-who")
```

Comment (hkx303) [reply tests/function/test_service.py:137]: Seems this would be better. What do you think of this?

**Review by hkx303 — APPROVED** [review]
ACK, pending small suggestion

File: tests/function/test_service.py (hunk 1 above)
Comment (Junefen) [reply tests/function/test_service.py]: updated to `ssh_guest.runcmd(f"ssh {server} -p {port} 'systemctl restart virt-who'")`
but after testing the `ssh_guest.runcmd(f"ssh {server} -p {port} systemctl restart virt-who")` also works.
thanks.

Comment (hkx303) [reply tests/function/test_service.py]: thanks for the quick reply

File: virtwho/base.py [hunk 2]
```diff
@@ -571,9 +572,7 @@ def ssh_access_no_password(ssh_local, ssh_remote, remote_host, remote_port=22):
         raise FailException("Failed to create ssh key ")
 
     # copy id_rsa.pup to remote host
-    ssh_remote.runcmd(
-        f"mkdir ~/.ssh/;" f"echo '{output}' >> ~/.ssh/authorized_keys"
-    )
+    ssh_remote.runcmd(f"mkdir ~/.ssh/;" f"echo '{output}' >> ~/.ssh/authorized_keys")
```
Comment (hkx303) [inline virtwho/base.py:575]:
```suggestion
    ssh_remote.runcmd(f"mkdir ~/.ssh/;echo '{output}' >> ~/.ssh/authorized_keys")
```

Comment (hkx303) [reply virtwho/base.py:575]: Sorry, also here I think :)

**Review by hkx303 — APPROVED** [review]
ACK, glad to see we have the PASS result

**Pull request merged**